test_data/asm/expand_fixed.asm

```diff
@@ -0,0 +1,141 @@
+use std::check::panic;
+
+enum Result<V, E> {
+    Ok(V),
+    Err(E)
+}
+
+/// A term in an array expression
+enum ArrayTerm {
+    Repeat(int[]),
+    Once(int[])
+}
+
+let MORE_THAN_ONE_REPEATED_ERROR = "unsolvable because more than one term is repeated";
+let NON_REPEATED_SIZE_EXCEEDS_DEGREE_ERROR = "non repeated array terms do not fit in degree";
+
+// returns the total size of the repeated array in this array expression
+let compute_length_of_repeated_part: ArrayTerm[], (-> int) -> Result<int, string> = |terms, degree| {
```

Comment:
could this take the degree directly instead of the function that returns the degree?